cl/search/management/commands/cl_dumpdata.py

```diff
@@ -34,42 +44,85 @@ def handle(self, *args, **options):
         fmt = options['format']
 
         self.stdout.write(
-            'Generating dump of up to %s randomly selected Opinions' % n
+            'Generating dump of up to %s randomly selected Opinions...' % n
         )
+
+        pks, cluster_pks = self._select_pks(n)
+        if n > len(pks):
+            self.stdout.write(
+                ' n > number of Opinions, serializing all (n=%s)' % len(pks)
+            )
+
+        try:
+            self._serialize(Opinion, fmt, Q(id__in=pks))
+
+            self._serialize(
+                OpinionsCited,
+                fmt,
+                Q(citing_opinion__in=pks) | Q(cited_opinion__in=pks)
+            )
+
+            self._serialize(OpinionCluster, fmt, Q(id__in=cluster_pks))
+            self._serialize(Docket, fmt, Q(clusters__in=cluster_pks))
+
+            self.stdout.write('Done!')
+
+        except Exception as e:
+            self.stderr.write('Failed to serialize: %s' % (e,))
+            if options['traceback']:
+                traceback.print_exc()
+
+    @staticmethod
+    def _select_pks(sample_size):
+        """
+        Select a random sampling of Opinions from the database.
+
+        Attributes:
+            sample_size -- number of Opinions to sample
+        """
+        n = sample_size
         pk_qs = Opinion.objects.values_list('id', flat=True)
```

Comment:
This query isn't great. It's probably not horrible, but it's not great. A different approach (the one taken by the old sample-data generator) could be to give up the guarantee that you get the exact number of objects you requested. If you do that, we can just get the total count of Opinions, and then take a sample of items between 1 and that number and use that in our `in` statements. 

Not all of the ids that we sampled would still be in the DB (due to deletions and whatnot), but it's within 10% or so last I checked. 

Or we could leave this problem for another day...